### DISCUSSION TITLE
OpenTelemetry API and SDK for Kotlin MPP

### DISCUSSION THREAD

**Discussion by JochenGuckSnk** [open-telemetry/community#958] [Community Discussions]
Hello Team,

Kotlin multi-platform (https://kotlinlang.org/docs/mpp-intro.html) is becoming more and more traction nowadays. It enables to program full stack applications. I am wondering if there are any plans to provide a API and SDK written in Kotlin?

I have already a port of the Tracing and Metrics SDK from OpenTelemetry-Jave in to Kotlin. 

- The Tracing SDK is working on JVM; JS and Native target.
- The Metrics SDK works only on JVM and Js. 
- Exporters are missing right now.

I wonder if there is any interest in using this as a starting point for an official OpenTelemetry-Kotlin SDK.

**Comment by JochenGuckSnk** [comment]
I just want to shear a small update. Metrics SDK is now also working on native. If u are interested in the code you can find it on Github https://github.com/dcxp/opentelemetry-kotlin. There are also maven packages.

**Reply by SergeyKanzhelev** [reply]
Can you register those in the registry? https://opentelemetry.io/registry/?s=kotlin&component=&language=

**Reply by JochenGuckSnk** [reply]
I created a pull request https://github.com/open-telemetry/opentelemetry.io/pull/1145 for adding the project to the repository. Just have to go through the CLA process from my side.
